frontend/src/lib-common/date.ts

```diff
@@ -1,25 +1,45 @@
-// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
 //
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
 import { format, isToday } from 'date-fns'
 
-export const DATE_FORMAT_DATE = 'dd.MM.yyyy'
 export const DATE_FORMAT_DATE_TIME = 'dd.MM.yyyy HH:mm'
-export const DATE_FORMAT_NO_YEAR = 'dd.MM.'
-export const DATE_FORMAT_SHORT_NO_YEAR = 'd.M.'
-export const DATE_FORMAT_ISO = 'yyyy-MM-dd'
-export const DATE_FORMAT_TIME_ONLY = 'HH:mm'
+
+type DateWithoutLeadingZeros = 'd.M.yyyy'
+type DateWithLeadingZeros = 'dd.MM.yyyy'
+
+type Time = 'HH:mm'
+type FullDate = DateWithLeadingZeros | DateWithoutLeadingZeros
+
+type DateWithoutYear = 'dd.MM.'
+type DateWithoutYearShort = 'd.M.'
+
+export type DateFormat = FullDate | DateWithoutYear | DateWithoutYearShort
+type DateTimeFormat = `${FullDate} ${Time}`
+
+type AllowedDateFormat = DateFormat | DateTimeFormat
 
 export function formatDate(
   date: Date | null | undefined,
-  dateFormat = DATE_FORMAT_DATE
+  dateFormat: AllowedDateFormat = 'dd.MM.yyyy'
 ): string {
   return date ? format(date, dateFormat) : ''
 }
 
+type IsoDate = 'yyyy-MM-dd'
+type IsoDateWithoutDay = 'yyyy-MM'
+type IsoFormat = IsoDate | IsoDateWithoutDay
+
+export function formatIsoDate(
+  date: Date | null | undefined,
+  isoFormat: IsoFormat = 'yyyy-MM-dd'
+): string {
+  return date ? format(date, isoFormat) : ''
+}
```

Comment:
I don't think we should even have this function since using `Date` instead of `LocalDate` shouldn't be encouraged when only the date matters.

Reply:
Let's see what I can do 👍